components/ws-manager-mk2/controllers/metrics.go

```diff
@@ -186,6 +185,7 @@ func (m *controllerMetrics) countWorkspaceStop(log *logr.Logger, ws *workspacev1
 	counter, err := m.totalStopsCounterVec.GetMetricWithLabelValues(reason, tpe, class)
 	if err != nil {
 		log.Error(err, "could not count workspace stop", "reason", "unknown", "type", tpe, "class", class)
+		return
```

Comment:
The only reason we can end up with an error here is when the number of arguments supplied doesn't match the counter cardinality. That in itself feels like a bug.

> GetMetricWithLabelValues returns the Counter for the given slice of label values (same order as the variable labels in Desc). If that combination of label values is accessed for the first time, a new Counter is created.

> It is possible to call this method without using the returned Counter to only create the new Counter but leave it at its starting value 0. See also the SummaryVec example.

> Keeping the Counter for later use is possible (and should be considered if performance is critical), but keep in mind that Reset, DeleteLabelValues and Delete can be used to delete the Counter from the CounterVec. In that case, the Counter will still exist, but it will not be exported anymore, even if a Counter with the same label values is created later.

> An error is returned if the number of label values is not the same as the number of variable labels in Desc (minus any curried labels).

> Note that for more than one label value, this method is prone to mistakes caused by an incorrect order of arguments. Consider GetMetricWith(Labels) as an alternative to avoid that type of mistake. For higher label numbers, the latter has a much more readable (albeit more verbose) syntax, but it comes with a performance overhead (for creating and processing the Labels map). See also the GaugeVec example.

I think we could simplify here, to avoid the scenario of an error entirely by doing

```
m.totalStopsCounterVec.WithLabelValues(...).Inc()
```

Reply:
Refactored